pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -602,28 +602,34 @@ self: super: {
   # https://github.com/athanclark/sets/issues/2
   sets = dontCheck super.sets;
 
-  # Install icons, metadata and cli program.
-  # Do not build hgettext as it is broken
-  # https://gitlab.freedesktop.org/bustle/bustle/issues/13
-  bustle = overrideCabal (disableCabalFlag (super.bustle.override { hgettext = null; }) "hgettext") (drv: {
-    buildDepends = [ pkgs.libpcap ];
-    buildTools = with pkgs.buildPackages; [ gettext perl help2man ];
-    patches = [
-      # fix build
-      # https://gitlab.freedesktop.org/bustle/bustle/merge_requests/14
-      (pkgs.fetchpatch {
-        url = "https://gitlab.freedesktop.org/bustle/bustle/commit/ee4b81cbc232d47ba9940f1987777b17452e71ff.patch";
-        sha256 = "0v9cvbmrma5jcqcg1narpm1549h0cg8mr6i00qxmq0x6hs04dnwa";
-      })
-      (pkgs.fetchpatch {
-        url = "https://gitlab.freedesktop.org/bustle/bustle/commit/aae6843f51f54679d440fb3813e61355dc8406b9.patch";
-        sha256 = "1a8hr38hd1gdkqhsy56hyl7njw8ci79iigr81aalkb7hn4ckvh2a";
-      })
-    ];
-    postInstall = ''
-      make install PREFIX=$out
-    '';
-  });
+  bustle = pkgs.lib.pipe super.bustle [
+    # Do not build hgettext as it is broken
+    # https://gitlab.freedesktop.org/bustle/bustle/issues/13
+    (bustle: bustle.override { hgettext = null; })
+    (bustle: disableCabalFlag bustle "hgettext")
+    (bustle: bustle.overrideAttrs (attrs: { meta = builtins.removeAttrs attrs.meta [ "hydraPlatforms" ]; }))
```

Comment:
I don't know anything about `bustle`, but my guess is that `hydraPlatforms` being set to `none` is a bug in `hackage2nix`.

`hydraPlatforms` is set to `none` automatically by `hackage2nix`/`cabal2nix` when the package is marked broken.  However, the `hydraPlatforms` setting should be removed automatically when the package is marked unbroken.

In this case, it appears that the package was marked unbroken, but the `hydraPlatforms` setting was not removed.

I feel like I've seen something like this happen before, but I don't remember the underlying cause.

Maybe @peti has an idea what could be going on here?

Reply:
I think I figured out what is going on here.

`bustle` has a license set as `OtherLicense`:

https://gitlab.freedesktop.org/bustle/bustle/blob/27ac836202028d1a9e4f18fff21d10aff802b04b/bustle.cabal#L8

However, bustle's actual license is some combination of LGPL and GPL:

https://gitlab.freedesktop.org/bustle/bustle/blob/27ac836202028d1a9e4f18fff21d10aff802b04b/LICENSE

`hackage2nix` sets `hydraPlatforms` to `none` by default for `OtherLicense`:

https://github.com/NixOS/cabal2nix/blob/b942b6aedd11852f6dc995bea944ca1c39ccc200/src/Distribution/Nixpkgs/Haskell/FromCabal.hs#L111

----------------------

So I think the correct solution here is to send a PR to `cabal2nix` overriding the `hdyraPlatforms` setting for `bustle`.

Here's the function this needs to be added to:

https://github.com/NixOS/cabal2nix/blob/b942b6aedd11852f6dc995bea944ca1c39ccc200/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L72-L77

@jtojnar Could you send this PR?  If you don't know Haskell, I can try to put together a PR for this.

Reply:
Hmm, looks like it no longer depends on dbus-core so upstream can switch to LGPL.

Reply:
Will be fixed in https://gitlab.freedesktop.org/bustle/bustle/merge_requests/17, I think the temporary fix is fine here.

Reply:
> In this case, it appears that the package was marked unbroken, but the `hydraPlatforms` setting was not removed.

The license might be a problem, indeed. Another problem is the system dependency on `gio-unix` which `hackage2nix` cannot resolve because we have no package of that name. We should probably add an appropriate name mapping to https://github.com/NixOS/cabal2nix/blob/master/src/Distribution/Nixpkgs/Haskell/FromCabal/Name.hs to remedy that issue. I can do that, but someone would have to tell me what that name should be mapped to, because I don't know.

Reply:
`gio-unix-2.0` is part of `glib` package.